lib/matplotlib/backends/backend_qt5agg.py

```diff
@@ -171,6 +198,10 @@ def __init__(self, figure):
             self.figure._original_dpi = self.figure.dpi
         self.figure.dpi = self._dpi_ratio * self.figure._original_dpi
 
+    def _update_figure_dpi(self):
```

Comment:
Can you use this in the `__init__` above as well (and get rid of the `hasattr`)?